constraint-solver/src/solver_tests.rs

```diff
@@ -0,0 +1,48 @@
+use crate::test_utils::*;
```

Comment:
Can you move this file to `constraint-solver/tests/solver.rs`?

Reply:
Or, since they are unit tests, just move them to a `test` sub-module in the `solver.rs` file.

Reply:
Hmm, I moved it to `constraint-solver/src/tests/solver.rs`. I think the `tests` directory is for integration tests? Happy to move it there too, but just checking.